### DISCUSSION TITLE
Is it possible to change the color of inline text in markdown?

### DISCUSSION THREAD

**Discussion by alexjlockwood** [squidfunk/mkdocs-material#5638] [Q&A]
For example, lets say that I wanted to make use of one of the material primary colors like "deep purple" defined [here](https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/#primary-color):

<img width="801" alt="Screenshot 2023-06-23 at 5 44 40 PM" src="https://github.com/squidfunk/mkdocs-material/assets/1380747/0ea63cdd-a376-478b-b418-097cd98c0814">

Is there a way to have part of markdown text take on this color value? Like:

```md
I want this text to be <span class="...">deep purple</span>
```

What is the preferred approach at doing something like this with mkdocs-material?

**Comment by squidfunk** [comment]
AFAIK, there are no explicit constructs to wrap a text with a `span` in Markdown, which is basically what you're asking for, so you can use `span` as you would use them in plain HTML. If you already have an inline element, e.g. a link, you can add classes using the [Attribute Lists](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#attribute-lists) extension:

``` markdown
[Link](#){ .my-css-class }
```

Now, if you want to use the colors as defined by Material for MkDocs, you can set `data-md-color-primary`, and then reuse the `--md-primary-fg-color` etc. variables. For the buttons, we're doing the following:

``` html
<button data-md-color-primary="pink"><code>pink</code></button>
```

``` css
button > code
    background-color: var(--md-primary-fg-color);
    color: var(--md-primary-bg-color);
    display: block;
```

This will use the pink color as defined by the theme for the button's background.
